front/app/modules/commercial/insights/hooks/useInsightsNetwork.ts

```diff
@@ -1,25 +1,69 @@
 import { useState, useEffect } from 'react';
 import {
   insightsNetworkStream,
-  IInsightsNetworkData,
+  IInsightsNetwork,
 } from '../services/insightsNetwork';
 
+import { interval } from 'rxjs';
+import { takeWhile, finalize } from 'rxjs/operators';
+
+import { API_PATH } from 'containers/App/constants';
+import streams from 'utils/streams';
+
+import { insightsTextNetworkAnalysisTasksStream } from 'modules/commercial/insights/services/insightsTextNetworkAnalysisTasks';
+
+const pollingStream = interval(3000);
+
 const useInsightsNetwork = (viewId: string) => {
+  const [loading, setLoading] = useState(true);
   const [insightsNetwork, setInsightsNetwork] = useState<
-    IInsightsNetworkData | undefined | null | Error
+    IInsightsNetwork | undefined | null | Error
   >(undefined);
 
   useEffect(() => {
     const subscription = insightsNetworkStream(viewId).observable.subscribe(
       (insightsNetwork) => {
-        setInsightsNetwork(insightsNetwork.data);
+        setInsightsNetwork(insightsNetwork);
       }
     );
 
-    return () => subscription.unsubscribe();
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [viewId]);
+
+  useEffect(() => {
```

Comment:
The polling is contained to the useNetwork hook in order to only poll when the Network component is mounted and ensure we are not putting unnecessary pressure on the server.